inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -629,3 +632,50 @@ def load_from_source_data(obj, eng):
         obj.save()
     except KeyError:
         raise ValueError("Can't start/restart workflow as 'source_data' is either missing or corrupted")
+
+
+def restart_workflow(wf, original_workflow, position=[0]):
+    """Restarts workflow
+
+    Args:
+        wf: Workflow to restart
+        original_workflow: Workflow which restarts
+        position: To which position wf should be restarted
+    """
+    wf.callback_pos = position
+    wf.extra_data['source_data']['extra_data']['delay'] = 10
+    restarted_by_wf_list = wf.extra_data.get('restarted-by-wf', [])
+    restarted_by_wf_list.append(original_workflow.id)
+    wf.extra_data['source_data']['extra_data']['restarted-by-wf'] = restarted_by_wf_list
+    wf.save()
+    db.session.commit()
+    wf.continue_workflow('restart_task', delayed=True)
+
+
+@with_debug_logging
+def run_next_if_necessary(obj, eng):
+    """Remove current wf id from matched workflows and run next one"""
+    blocked_wfs = get_wfs_not_completed(obj, False)
+    next_wf = None
+    for wf_id in blocked_wfs:
+        wf = workflow_object_class.get(wf_id)
+        if obj.id in wf.extra_data.get('holdingpen_matches', []):
+            wf.extra_data['holdingpen_matches'].remove(obj.id)
+            wf.save()
+        if not wf.extra_data.get('holdingpen_matches', []) and not next_wf:
+            # If workflow is not blocked by any other workflow
+            # And there is no next workflow set
+            # then set is as next to restart
+            next_wf = wf
+    db.session.commit()
+    if next_wf:
+        restart_workflow(next_wf, obj)
+
+
+@with_debug_logging
+def delay_if_necessary(obj, eng):
+    """Delays workflow if necessary,
+    delay timeout should be in `extra_data['source_data']['persistant_data']['delay']` key """
+    delay = get_value(obj.extra_data, 'source_data.extra_data.delay')
+    if delay:
+        time.sleep(float(delay))
```

Comment:
why the `float`?

Reply:
as sleep accepts 1 and 1.5 and 1.9432 but it won't accept 'an hour' etc. so it's good to cast value to float (as we won't reduce accuracy), but if there will be something not castable to float it will produce more readable exception in wf _error_msg ("ValueError: could not convert string to float: xxxx) than just `TypeError: a float is required`

Reply:
I'd just ``time.sleep(delay)`` ☺️

Reply:
Which wf `_error_msg` is easier to debug : `TypeError: a float is required` or `ValueError: could not convert string to float: xxxx`?